dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:stack_trace/stack_trace.dart';
+
+import 'protocol/sentry_stack_frame.dart';
+
+class SentryStackTraceFactory {
+  const SentryStackTraceFactory();
+
+  List<SentryStackFrame> getStackFrames(dynamic stackTrace) {
+    if (stackTrace == null) return [];
+
+    final chain = stackTrace is StackTrace
+        ? Chain.forTrace(stackTrace)
+        : Chain.parse(stackTrace as String);
+
+    final frames = <SentryStackFrame>[];
+    for (var t = 0; t < chain.traces.length; t += 1) {
+      final encodedFrames =
+          chain.traces[t].frames.map((f) => encodeStackTraceFrame(f));
+
+      frames.addAll(encodedFrames);
+
+      if (t < chain.traces.length - 1) {
+        frames.add(SentryStackFrame.asynchronousGapFrameJson);
+      }
+    }
+
+    return frames.reversed.toList();
+  }
+
+  SentryStackFrame encodeStackTraceFrame(Frame frame) {
+    final sentryStackFrame = SentryStackFrame()
+      ..absPath = '${_absolutePathForCrashReport(frame)}'
+      ..function = frame.member
+      ..lineNo = frame.line
+      ..colNo = frame.column
+      ..inApp = !frame.isCore;
```

Comment:
See: https://github.com/getsentry/sentry-dart/pull/155/files#r517188682

we can still use `frame.isCore` in conjunction with inAppExcudes/Includes, but this affects grouping, one might not want to consider 3rd party libs for grouping exceptions together.

Reply:
✅